wgsl/index.bs

```diff
@@ -17809,40 +17809,98 @@ fn atomicLoad(atomic_ptr: ptr<AS, atomic<T>, read_write>) -> T
+
+Each function performs the following steps atomically:
+
+1. Load the original value pointed to by `atomic_ptr`.
+2. Obtains a new value by performing the operation (e.g. max) from the function
+    name with the value |v|.
+3. Store the new value using `atomic_ptr`.
+
+Each function returns the original value stored in the atomic object before the operation.
+
+#### `atomicAdd` #### {#atomic-add}
 
 ```wgsl
 fn atomicAdd(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs an addition operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+#### `atomicSub` #### {#atomic-sub}
+
+```wgsl
 fn atomicSub(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs a subtraction operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+#### `atomicMax` #### {#atomic-max}
+
+```wgsl
 fn atomicMax(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs a maximum operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+#### `atomicMin` #### {#atomic-min}
+
+```wgsl
 fn atomicMin(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs a minimum operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+#### `atomicAnd` #### {#atomic-and}
+
+```wgsl
 fn atomicAnd(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs a bitwise AND operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+#### `atomicOr` #### {#atomic-or}
+
+```wgsl
 fn atomicOr(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs a bitwise OR operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+#### `atomicXor` #### {#atomic-xor}
+
+```wgsl
 fn atomicXor(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
 ```
-Each function performs the following steps atomically:
 
-1. Load the original value pointed to by `atomic_ptr`.
-2. Obtains a new value by performing the operation (e.g. max) from the function
-    name with the value |v|.
-3. Store the new value using `atomic_ptr`.
+Atomically performs a bitwise XOR operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
 
-Each function returns the original value stored in the atomic object.
+#### `atomicExchange` #### {#atomic-exchange}
```

Comment:
`atomicExchange` and `atomicCompareExchangeWeak` are both RMW operations, but the base section describes the steps taken and these don't fit as neatly. Exchange would be a store, but we don't use that as an expression name or builtin function. I wonder if these should be pulled up a level in the hierarchy and RMW subsection be renamed to RMW arithmetic and logical operations.

As above though, the polyfills could be quite useful for these exchanges as explanation of functionality.

Reply:
I've pulled these up a level, however, is it OK to defer polyfills for these? I felt like examples might work better for these ones.